### PR TITLE
[incubator/kafka] updated default version to latest release

### PR DESCRIPTION
**Pull request by Comradin**
The latest bugfix release for confluent kafka is 4.1.2-2

What this PR does / why we need it:
The confluent kafka release 4.1.2 fixes some important issues
https://docs.confluent.io/4.1.2/release-notes.html#id5

The most important one, from my perspective, is "KAFKA-6855" which addresses an startup issue with Java 10.

Which issue this PR fixes
None, this is only an update for better default values.

Special notes for your reviewer:
Nothing has been changed from the operations view of this chart. Just a release update.

### CODE REVIEW

File: incubator/kafka/README.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ following configurable parameters:
...
 |------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------|
 | `image`                                        | Kafka Container image name                                                                                                                                               | `confluentinc/cp-kafka`                                            |
-| `imageTag`                                     | Kafka Container image tag                                                                                                                                                | `4.0.0`                                                            |
+| `imageTag`                                     | Kafka Container image tag                                                                                                                                                | `4.1.2-2`                                                            |
```
Comment (davidkarlsen) [inline incubator/kafka/README.md:60]: Chart.yaml needs versionbump too, please also update the appVersion as well.

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: incubator/kafka/README.md (hunk 1 above)
Comment (Comradin) [reply incubator/kafka/README.md:60]: Oh, sure, thanks for the notice

**Comment by benjigoldberg** [comment]
@Comradin Thanks for the PR! As @davidkarlsen mentioned you will also need to bump the `version` to `0.8.8` in addition to the bump to `appVersion` that you've already made. `appVersion` tracks the version of Kafka, and `version` tracks the version of the Kafka chart -- so we need to increment both.

**Comment by davidkarlsen** [comment]
@Comradin also the CLA needs to be signed

**Comment by maver1ck** [comment]
Why not bump version to 5.0.0 ?

**Comment by Comradin** [comment]
@davidkarlsen I signed the CLA today and sent the pdf to cla@cncf.io this afternoon CEST.

@maver1ck I haven't testet this chart with 5.0.0 / Kafka 2.0 yet. Next PR I'll raise to 5.0.0

**Comment by Comradin** [comment]
Sorry, did not realize one can sign the CLA via the electronic form .. sorry for the mess

**Comment by Comradin** [comment]
Hi, any news on this?

**Comment by benjigoldberg** [comment]
/lgtm

Thanks for the contribution @Comradin

**Pull request merged**